app/src/main/java/io/outblock/lilico/manager/account/AccountManager.kt

```diff
@@ -123,30 +142,52 @@ object AccountManager {
             callback(false)
             return
         }
-        val wallet = AccountWalletManager.getHDWalletByUID(account.wallet?.id ?: "")
-        if (wallet == null) {
-            callback(false)
-            return
-        }
         val deviceInfoRequest = DeviceInfoManager.getDeviceInfoRequest()
         val service = retrofit().create(ApiService::class.java)
-        val resp = service.login(
-            LoginRequest(
-                signature = wallet.sign(
-                    getFirebaseJwt()
-                ),
-                accountKey = AccountKey(publicKey = wallet.getPublicKey(removePrefix = true)),
-                deviceInfo = deviceInfoRequest
+        val prefix = account.prefix
+        val resp: LoginResponse
+        if (prefix == null) {
+            val wallet = AccountWalletManager.getHDWalletByUID(account.wallet?.id ?: "")
+            if (wallet == null) {
+                callback(false)
```

Comment:
Better return error, instead of false.

Use sealed class, Success, and Error(code, message) or Error enum